api/src/test/java/org/openmrs/ui/framework/formatter/FormatterServiceTest.java

```diff
@@ -49,25 +47,23 @@ public String toString() {
 	}
 	
 	@Test
-	@DirtiesContext
 	public void testMessage() throws Exception {
 		MessageSource messageSource = mock(MessageSource.class);
-		
+		FormatterService messageFormatService = new FormatterService();
+		messageFormatService.setMessageSource(messageSource);
 		HandlebarsFormatterFactory classFormatter = new HandlebarsFormatterFactory();
 		classFormatter.setForClass("org.openmrs.Obs");
 		classFormatter.setTemplate("{{ message 'testing.123.testing' }} something");
-		formatterService.addClassFormatter(classFormatter);
```

Comment:
Why did you get rid of the above?

Reply:
I didn't. I just changed the variable. From `formatterService` to `messageFormatService`

Reply:
It is not easy to tell what exactly changed or what was necessary to make the test pass.
For instance, why do we now instantiate the message format service like `FormatterService messageFormatterService = new FormatterService()`

Reply:
The error I mentioned above happens because of the test leakage from the formatterService. That's why I added the method variable for it.

Reply:
Can you explain more what you mean by the test leackage?

Reply:
Because we removed @DirtiesContext changes happen to the `formatterService` on the other two tests passed on to this test. This causes the test leakage. That's why I decided to use a separate FormatterService for this test.